### PR TITLE
add helper for computing noise_multiplier from epsilon

### PR DESCRIPTION
**Pull request by jagielski**
A common workflow starts from a privacy budget, and requires obtaining the noise multiplier that is required for DP-SGD to satisfy the given privacy budget. With the current codebase, this requires running a binary search. This pull request implements this binary search. It can be found in the analysis/compute_noise_from_budget.py script, using the same API as the existing analysis/compute_dp_sgd_privacy.py.

### CODE REVIEW

File: tensorflow_privacy/privacy/analysis/compute_noise_from_budget.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+r"""Command-line script for computing privacy of a model trained with DP-SGD.
+
+The script applies the RDP accountant to estimate privacy budget of an iterated
+Sampled Gaussian Mechanism. The mechanism's parameters are controlled by flags.
+
+Example:
+  compute_dp_sgd_privacy
```
Comment (schien1729) [inline tensorflow_privacy/privacy/analysis/compute_noise_from_budget.py:21]: Please update this command line to reflect name of program.

File: tensorflow_privacy/privacy/analysis/compute_noise_from_budget_lib.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  eps, _, opt_order = get_privacy_spent(orders, rdp, target_delta=delta)
+
+  return eps, opt_order
+
+
+def compute_noise(n, batch_size, target_epsilon, epochs, delta):
+  """Compute noise based on the given hyperparameters."""
+  q = batch_size / n  # q - the sampling ratio.
+  if q > 1:
+    raise app.UsageError('n must be larger than the batch size.')
```
Comment (schien1729) [inline tensorflow_privacy/privacy/analysis/compute_noise_from_budget_lib.py:52]: Minor note: Given that n is usually fixed, while the batch size can be varied, maybe rephrase this to say that batch size must be at most n?

File: tensorflow_privacy/privacy/analysis/compute_noise_from_budget.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+FLAGS = flags.FLAGS
+
+flags.DEFINE_integer('N', None, 'Total number of examples')
+flags.DEFINE_integer('batch_size', None, 'Batch size')
+flags.DEFINE_float('epsilon', None, 'Target epsilon for DP-SGD')
+flags.DEFINE_float('epochs', None, 'Number of epochs (may be fractional)')
+flags.DEFINE_float('delta', 1e-6, 'Target delta')
+
+
+def main(argv):
```
Comment (schien1729) [inline tensorflow_privacy/privacy/analysis/compute_noise_from_budget.py:55]: As I read it, in the rare case that noise of 1e-5 is too much, there's not much the user can do to use this tool. Can you either document that, or maybe add a minimum noise argument to compute_noise, defaulting to 1e-5?

**Review by schien1729 — CHANGES_REQUESTED** [review]
Thanks, Matthew! This is indeed a useful tool that users have requested. I've made some minor suggestions-- sorry it took so long.

**Comment by jagielski** [comment]
Thanks for the comments, Steve! I've addressed each of your comments in the latest commit. On the last one, I've elected to add a min_noise parameter, and I've added a test to check that if min_noise is too large, it has the expected behavior (just return 0).

**Review by schien1729 — APPROVED** [review]
